scripts/cherry-pick-patch-commits.js

```diff
@@ -0,0 +1,164 @@
+async function getMostRecentTag() {
+  const tags = await simpleGit.tags();
+  // Filter old independent-versioned tags and pre-releases
+  const filteredTags = tags.all.filter((tag) => /^v/.test(tag) && !/-/.test(tag));
+  return filteredTags[filteredTags.length - 1];
+}
+
+// Resolves to an array of commits after the given tag, from earliest to latest (for proper cherry-picking).
+async function getCommitsAfterTag(tag) {
+  if (!args.includes('--no-fetch')) {
+    await simpleGit.fetch();
+  }
+  const log = await simpleGit.log({
+    from: tag,
+    to: 'origin/master',
+    // simple-git defaults to only including subject + tag/branch (%s%d) in message, which adds noise and is not useful
+    // for finding BREAKING CHANGEs (which are in body, %b).
+    format: {
+      hash: '%H', // Contains full hash only, for cherry-picking
+      message: '%s\n\n%b', // Contains subject + body for passing to conventional-commits-parser
+      oneline: '%h %s' // Contains abbreviated hash + subject for summary lists
+    },
+    // Use a splitter that is very unlikely to be included in commit descriptions
+    splitter: ';;;'
+  });
+  return log.all.reverse();
+}
+
+// Returns true if commit should NOT be cherry-picked.
+function shouldSkipCommit(logLine) {
+  const parsedCommit = parser.sync(logLine.message, parserOpts);
+  return parsedCommit.type === 'feat' || // feature commit
+    parsedCommit.notes.find((note) => note.title === 'BREAKING CHANGE') || // breaking change commit
+    (parsedCommit.type === 'chore' && parsedCommit.subject === 'Publish'); // Publish (version-rev) commit
+}
+
+// Checks out the given tag and attempts to cherry-pick each commit in the list against it.
+// If a conflict is encountered, that cherry-pick is aborted and processing moves on to the next commit.
+async function attemptCherryPicks(tag, list) {
+  const results = {
+    successful: [],
+    conflicted: [],
+    skipped: [],
+  }
+
+  console.log(`Checking out ${tag}`)
+  await simpleGit.checkout([tag]);
+
+  for (let logLine of list) {
+    if (shouldSkipCommit(logLine)) {
+      results.skipped.push(logLine);
+      continue;
+    }
+
+    try {
+      await simpleGit.raw(['cherry-pick', '-x', logLine.hash]);
+      results.successful.push(logLine);
+    } catch (e) {
+      if (e.message.includes(CONFLICT_MESSAGE)) {
+        results.conflicted.push(logLine);
+        await simpleGit.raw(['cherry-pick', '--abort']);
+      } else {
+        console.error(`${logLine.hash} unexpected failure!`, e);
+      }
+    }
+  }
+
+  return results;
+}
+
+// Given a string containing a command and arguments, runs the command and returns true if it exits successfully.
+// The command's I/O is piped through to the parent process intentionally to be visible in the console.
+function checkSpawnSuccess(command) {
+  const parts = command.split(' ');
+  return spawnSync(parts[0], parts.slice(1), {stdio: 'inherit'}).status === 0;
```

Comment:
The `parts.slice(1)` is because `spawnSync` takes an array?

Reply:
Yeah, it takes a command + an array of arguments to that command.